src/ovs_events.c

```diff
@@ -0,0 +1,631 @@
+  ovs_events_iface_list_t *del_iface = NULL;
+  sfree(ovs_events_ctx.ovs_db_select_params);
+  while (ovs_events_ctx.config.ifaces) {
+    del_iface = ovs_events_ctx.config.ifaces;
+    ovs_events_ctx.config.ifaces = ovs_events_ctx.config.ifaces->next;
+    sfree(del_iface);
+  }
+}
+
+/* Parse plugin configuration file and store the config
+ * in allocated memory. Returns negative value in case of error.
+ */
+static int ovs_events_plugin_config(oconfig_item_t *ci) {
+  ovs_events_iface_list_t *new_iface;
+
+  for (int i = 0; i < ci->children_num; i++) {
+    oconfig_item_t *child = ci->children + i;
+    if (strcasecmp("SendNotification", child->key) == 0) {
+      if (cf_util_get_boolean(child,
+                              &ovs_events_ctx.config.send_notification) != 0)
+        OVS_EVENTS_CONFIG_ERROR(child->key);
+    } else if (strcasecmp("Address", child->key) == 0) {
+      if (cf_util_get_string_buffer(
+              child, ovs_events_ctx.config.ovs_db_node,
+              sizeof(ovs_events_ctx.config.ovs_db_node)) != 0)
+        OVS_EVENTS_CONFIG_ERROR(child->key);
+    } else if (strcasecmp("Port", child->key) == 0) {
+      if (cf_util_get_string_buffer(
+              child, ovs_events_ctx.config.ovs_db_serv,
+              sizeof(ovs_events_ctx.config.ovs_db_serv)) != 0)
+        OVS_EVENTS_CONFIG_ERROR(child->key);
+    } else if (strcasecmp("Socket", child->key) == 0) {
+      if (cf_util_get_string_buffer(
+              child, ovs_events_ctx.config.ovs_db_unix,
+              sizeof(ovs_events_ctx.config.ovs_db_unix)) != 0)
+        OVS_EVENTS_CONFIG_ERROR(child->key);
+    } else if (strcasecmp("Interfaces", child->key) == 0) {
+      for (int j = 0; j < child->values_num; j++) {
+        /* check value type */
+        if (child->values[j].type != OCONFIG_TYPE_STRING) {
+          ERROR(OVS_EVENTS_PLUGIN
+                ": given interface name is not a string [idx=%d]",
+                j);
+          goto failure;
+        }
+        /* allocate memory for configured interface */
+        if ((new_iface = malloc(sizeof(*new_iface))) == NULL) {
+          ERROR(OVS_EVENTS_PLUGIN ": malloc () copy interface name fail");
+          goto failure;
+        } else {
+          /* store interface name */
+          sstrncpy(new_iface->name, child->values[j].value.string,
+                   sizeof(new_iface->name));
+          new_iface->next = ovs_events_ctx.config.ifaces;
+          ovs_events_ctx.config.ifaces = new_iface;
+          DEBUG(OVS_EVENTS_PLUGIN ": found monitored interface \"%s\"",
+                new_iface->name);
+        }
+      }
+    } else {
+      ERROR(OVS_EVENTS_PLUGIN ": option '%s' is not allowed here", child->key);
+      goto failure;
+    }
+  }
+  return (0);
+
+failure:
+  ovs_events_config_free();
+  return (-1);
+}
+
+/* Dispatch OVS interface link status event to collectd */
+static void
+ovs_events_dispatch_notification(const ovs_events_iface_info_t *ifinfo) {
```

Comment:
This is done by clang-format. I've addressed it anyway but I suppose it will be changed back during next clan-format running :)